### PR TITLE
Add some configuration options and document them

### PR DESCRIPTION
**Pull request by robsears**

### CODE REVIEW

**Comment by nz** [comment]
Maybe worth noting that it's possible to provide an ES URL via environment variable. We're supporting BONSAI_URL and ELASTICSEARCH_URL.

File: README.md [hunk 1]
```diff
@@ -22,7 +22,14 @@ Or install it yourself as:
 
 ## Usage
 
-TODO: Write usage instructions here
+```
+elasticsearch:
+  url: "http://localhost:9200/" # Required. Supports auth and SSL: https://user:pass@someurl.com
+  number_of_shards: 1           # Optional. Default is 1 primary shard.
+  number_of_replicas: 1         # Optional. Default is 0 replicas.
```
Comment (nz) [inline README.md]: We start with 0 replicas for the initial bulk update, but should update to this value when we're done.

File: lib/searchyou/indexer.rb [hunk 2]
```diff
@@ -29,11 +29,30 @@ def working?
...
+    end
+
+    # User configurable setting for primary shards
+    def number_of_shards
+      site.config['elasticsearch']['number_of_shards'] || 1
+    end
+
+    # User configurable setting for replicas
+    def number_of_replicas
+      site.config['elasticsearch']['number_of_replicas'] || 0
```
Comment (nz) [inline lib/searchyou/indexer.rb]: Let's default this to 1 for the "production" setting after finalizing.

File: lib/searchyou/indexer.rb [hunk 3]
```diff
@@ -29,11 +29,30 @@ def working?
       working || queue.length > 0
     end
 
+    # User configurable prefix for the index.
+    def es_index_prefix
+      site.config['elasticsearch']['index_name'] || "jekyll"
```
Comment (nz) [inline lib/searchyou/indexer.rb]: `site` doesn't exist as an instance attribute anymore ;)

**Comment by robsears** [comment]
Closing. I can do this another way.

**Pull request closed without merging**